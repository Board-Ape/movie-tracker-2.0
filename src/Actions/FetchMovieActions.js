export default function FetchMovieActions(url) {
  return dispatch => {

    dispatch({ type: 'MOVIES_IS_LOADING' });

    return fetch(`${url}`)
      .then(res => res.json())
      .then(movies => { 
        return dispatch({ type: 'MOVIES_FETCH_DATA_SUCCESS', payload: movies });
      })
      .catch( () => {
        return dispatch({ type: 'MOVIES_HAS_ERRORED' });
      });

  };
}