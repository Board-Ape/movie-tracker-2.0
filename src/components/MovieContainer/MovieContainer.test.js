import React from 'react';
import  { 
  MovieContainer, 
  mapStateToProps, 
  mapDispatchToProps 
} from './MovieContainer';
import { shallow } from 'enzyme';
import * as actions from '../../Actions';
import { Provider } from 'react-redux';

describe('MovieContainer tests', () => {
  let movieContainer;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      location: {},
      movies: [{ title: "Movie" }],
      hasErrored: true,
      isLoading: true,
      favorites: [{ title: "Movie" }]
    };
    movieContainer = shallow(
      <Provider><MovieContainer {...mockProps} /></Provider>
    );
  });


  it('should match the snapshot', () => {
    expect(movieContainer).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(movieContainer).toBeDefined();
  });

  it.skip('should pull movies from the store', () => {
    const mockStore = {
      movies: [{ title: "Movie" }],
      favorites: [{ title: "Movie" }]
    };
    
    const result = mapStateToProps(mockStore);
    expect(result.movies).toEqual(mockStore.movies);
    expect(result.favorites).toEqual(mockStore.favorites);
  });



});





// describe('MovieContianer', () => {

//   describe('componentDidMount', () => {

//     it('sets the state componentDidMount', async () => {
//       window.fetch = jest.fn().mockImplementation(() => ({
//         status: 200,
//         json: () => new Promise((resolve, reject) => {
//           resolve({
//             movies: [
//               { title: 'Star Wars', score: 10 }, { title: 'Blade Runner', score: 3 }
//             ]
//           })
//         })
//       }))

//       const renderedComponent = await shallow(<MovieContainer />);
//       await renderedComponent.update();
//       expect(renderedComponent('movies').length).toEqual(2);
//     })
//   })  
// })