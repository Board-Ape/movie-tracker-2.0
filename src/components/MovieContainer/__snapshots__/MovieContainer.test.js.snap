// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieContainer
    favorites={
        Array [
            Object {
              "id": 12345,
              "title": "Movie",
            },
          ]
    }
    fetchMovieList={[Function]}
    hasErrored={false}
    isLoading={false}
    location={Object {}}
    movies={
        Object {
            "movies": Array [
              Object {
                "id": 57203,
                "title": "Movie",
              },
            ],
          }
    }
    userData={
        Object {
            "error": false,
            "signedIn": true,
            "userData": Object {
              "email": "123",
              "id": 2,
              "name": "123",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(MovieCard)
          movie={
                    Object {
                              "id": 57203,
                              "title": "Movie",
                            }
          }
          overview={undefined}
          poster={undefined}
          rating={undefined}
          title="Movie"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "57203",
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "id": 57203,
            "title": "Movie",
          },
          "overview": undefined,
          "poster": undefined,
          "rating": undefined,
          "title": "Movie",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(MovieCard)
            movie={
                        Object {
                                    "id": 57203,
                                    "title": "Movie",
                                  }
            }
            overview={undefined}
            poster={undefined}
            rating={undefined}
            title="Movie"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "57203",
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "id": 57203,
              "title": "Movie",
            },
            "overview": undefined,
            "poster": undefined,
            "rating": undefined,
            "title": "Movie",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
